src/drivers/distance_sensor/ll40ls/LidarLiteI2C.cpp

```diff
@@ -380,16 +380,13 @@ LidarLiteI2C::collect()
 	}
 
 	uint8_t ll40ls_signal_strength = val[0];
-
-	uint8_t signal_min = 0;
-	uint8_t signal_max = 0;
-	uint8_t signal_value = 0;
+	uint8_t signal_quality;
 
 	// We detect if V3HP is being used
 	if (_is_v3hp) {
-		signal_min = LL40LS_SIGNAL_STRENGTH_MIN_V3HP;
-		signal_max = LL40LS_SIGNAL_STRENGTH_MAX_V3HP;
-		signal_value = ll40ls_signal_strength;
+		//Normalize signal strength to 0...100 percent using the absolute signal strength.
+		signal_quality = 100 * math::max(ll40ls_signal_strength - LL40LS_SIGNAL_STRENGTH_MIN_V3HP, 0) /
+				 (LL40LS_SIGNAL_STRENGTH_MAX_V3HP - LL40LS_SIGNAL_STRENGTH_MIN_V3HP);
 
```

Comment:
@BazookaJoe1900 , since the `(LL40LS_SIGNAL_STRENGTH_MAX_V3HP - LL40LS_SIGNAL_STRENGTH_MIN_V3HP)` calculation is being done in a few places within the collect() method in this PR, could you add this to the header file as something like `static const int LL40LS_SIGNAL_MARGIN {LL40LS_SIGNAL_STRENGTH_MAX_V3HP - LL40LS_SIGNAL_STRENGTH_MIN_V3HP};` (It should save a few CPU cycles and makes reading the code easy.)

Reply:
@mcsauder I don't see that `(LL40LS_SIGNAL_STRENGTH_MAX_V3HP - LL40LS_SIGNAL_STRENGTH_MIN_V3HP)` is done in multiple places, can you re-check?
I think that the current readability is quite clear.
any way, thous are two consts, and in most ways the optimizer will do pretty good optimization, so no worry about CPU cycle there...
BTW, I re-based the code, and made some more small change.
about the driver, its optimization and other updates, It might be done later on other PRs, for now it works for me ok.

Reply:
I read too quickly. You're correct. Thanks @BazookaJoe1900 .

Reply:
@mcsauder note that I fixed error few minutes ago on the second commit